src/GuardClauses/GuardClauseExtensions.cs

```diff
@@ -23,12 +23,13 @@ public static class GuardClauseExtensions
         /// <param name="guardClause"></param>
         /// <param name="input"></param>
         /// <param name="parameterName"></param>
+        /// <param name="message">Optional. Custom error message</param>
         /// <returns><paramref name="input" /> if the value is not null.</returns>
-        public static T Null<T>([JetBrainsNotNull] this IGuardClause guardClause, [NotNull, JetBrainsNotNull][ValidatedNotNull] T input, [JetBrainsNotNull] string parameterName)
+        public static T Null<T>([JetBrainsNotNull] this IGuardClause guardClause, [NotNull, JetBrainsNotNull][ValidatedNotNull] T input, [JetBrainsNotNull] string parameterName, string? message = null)
         {
             if (input is null)
             {
-                throw new ArgumentNullException(parameterName);
+                throw new ArgumentNullException(parameterName, message);
```

Comment:
This changes the behavior of guard clauses that don't pass in a message. Normally there would be a default message:

https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/ArgumentNullException.cs#L33

Now that constructor is bypassed. I think you need an if check to see if message is provided and then throw the appropriate exception created with or without the message argument.

Reply:
My interpretation on the above would be to : `throw new ArgumentNullException(message, (Exception?) null);` when the message is supplied.

Reply:
Will that overload end up using the same default ArgumentNullException text from the base class? If so, that sounds good (I haven't had time to trace it down).

Another unit test you could write would be to verify that the guard clause does, in fact, generate the same exception as you get with just `new ArgumentNullException()`.